components/compiler-families/llvm-compilers/SPECS/llvm-compilers.spec

```diff
@@ -0,0 +1,191 @@
+#----------------------------------------------------------------------------bh-
+# This RPM .spec file is part of the OpenHPC project.
+#
+# It may have been modified from the default version supplied by the underlying
+# release package (if available) in order to apply patches, perform customized
+# build/install configurations, and supply additional files to support
+# desired integration conventions.
+#
+#----------------------------------------------------------------------------eh-
+
+%include %{_sourcedir}/OHPC_macros
+
+%global gnuver 6.3.0
+%global cmakemajor 3.8
+%global cmakever 3.8.1
+
+%define pname llvm-compilers
+
+Summary:   The LLVM Compiler Infrastructure
+Name:      %{pname}%{PROJ_DELIM}
+Version:   4.0.0
+Release:   1
+License:   UIUC
+Group:     %{PROJ_NAME}/compiler-families
+URL:       http://www.llvm.org
+BuildRoot: %{_tmppath}/%{pname}-%{version}-%{release}-root
+Source0:   http://releases.llvm.org/%{version}/llvm-%{version}.src.tar.xz
+Source1:   http://releases.llvm.org/%{version}/cfe-%{version}.src.tar.xz
+Source2:   http://releases.llvm.org/%{version}/compiler-rt-%{version}.src.tar.xz
+Source3:   http://releases.llvm.org/%{version}/libcxx-%{version}.src.tar.xz
+Source4:   http://releases.llvm.org/%{version}/libcxxabi-%{version}.src.tar.xz
+Source5:   http://releases.llvm.org/%{version}/libunwind-%{version}.src.tar.xz
+Source6:   http://releases.llvm.org/%{version}/lld-%{version}.src.tar.xz
+Source7:   http://releases.llvm.org/%{version}/openmp-%{version}.src.tar.xz
+Source8:   https://cmake.org/files/v%{cmakemajor}/cmake-%{cmakever}.tar.gz
+Source9:   OHPC_macros
+Source10:  OHPC_setup_compiler
+Patch1:    openmp-4.0.0-gsupport.patch
+
+%define debug_package %{nil}
+
+BuildRequires: gnu-compilers%{PROJ_DELIM}
+BuildRequires: make
+BuildRequires: perl
+BuildRequires: python
+BuildRequires: pkgconfig
+BuildRequires: binutils-devel
+Requires:      binutils
+Requires:      gnu-compilers%{PROJ_DELIM} = %{gnuver}
```

Comment:
I did the following:

```--- a/components/OHPC_macros
+++ b/components/OHPC_macros
@@ -77,6 +77,10 @@ Requires:      gnu-dts6-compilers%{PROJ_DELIM}
 BuildRequires: gnu-7-compilers%{PROJ_DELIM}
 Requires:      gnu-7-compilers%{PROJ_DELIM}
 %endif
+%if "%{compiler_family}" == "llvm"
+BuildRequires: llvm-compilers%{PROJ_DELIM}
+Requires:      llvm-compilers%{PROJ_DELIM}
+%endif
 
 %endif
 
```

```
--- a/components/compiler-families/llvm-compilers/SPECS/llvm-compilers.spec
+++ b/components/compiler-families/llvm-compilers/SPECS/llvm-compilers.spec
@@ -8,6 +8,7 @@
 #
 #----------------------------------------------------------------------------eh-
 
+%define ohpc_compiler_dependent 1
 %include %{_sourcedir}/OHPC_macros
 
 %global gnuver 6.3.0
```
This will pull in the compiler define in compiler_family.

Github's code reviews are sometimes hard to read. Following lines can be removed

```
-Source10:  OHPC_setup_compiler
-%define debug_package %{nil}
-
-BuildRequires: gnu-compilers%{PROJ_DELIM}
-Requires:      gnu-compilers%{PROJ_DELIM} = %{gnuver}
```

And this change:

```
@@ -100,8 +96,7 @@ ln -s cmake-%{cmakever} cmake
 %build
 
 %install
-export OHPC_COMPILER_FAMILY=gnu
-. %{_sourcedir}/OHPC_setup_compiler
+ %ohpc_setup_compiler
 
 mkdir build-cmake
 cd build-cmake
```